repository/src/interpret/push.rs

```diff
@@ -0,0 +1,113 @@
+use super::*;
+
+pub async fn test_push_eligibility(
+    raw: &RawRepository,
+    commit_hash: CommitHash,
+    branch_name: String,
+    timestamp: Timestamp,
+    signature: TypedSignature<(CommitHash, String, Timestamp)>,
+    _timestamp_to_test: Timestamp,
+) -> Result<bool, Error> {
+    let reserved_state = raw.read_reserved_state().await?;
+    // TODO: elaborate this to consider `Member::expelled`.
+    let is_eligible = reserved_state
+        .members
+        .iter()
+        .any(|member| member.public_key == *signature.signer());
+    let is_eligible = is_eligible
+        && signature
+            .verify(&(commit_hash, branch_name, timestamp))
+            .is_ok();
+    // TODO: put the threshold in the config.
+    // let is_eligible = is_eligible && (timestamp_to_test - timestamp).abs() <= 1000;
+    Ok(is_eligible)
+}
+
+pub async fn broadcast(
+    raw: &mut RawRepository,
+    private_key: Option<PrivateKey>,
+) -> Result<(), Error> {
+    let agendas = read_agendas(raw).await?;
+    let blocks = read_blocks(raw).await?;
+    let remotes = raw.list_remotes().await?;
```

Comment:
Could you add `read_governance_approved_agendas()` and implement it? (currently it's todo, but I'm sure it's easy)

Reply:
Implemented.